src/components/WorkerStatusBadge.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <template v-if="worker">
+    <p-tag class="worker-status-badge">
+      <WorkerStatusIcon :status="workerStatus" class="worker-status-badge__icon" />
+      {{ label }}
+    </p-tag>
+  </template>
+</template>
+
+<script lang="ts" setup>
+  import { computed } from 'vue'
+  import WorkerStatusIcon from '@/components/WorkerStatusIcon.vue'
+  import { WorkPoolWorker } from '@/models'
+  import { capitalize } from '@/utilities'
+
+  const props = defineProps<{
+    worker: WorkPoolWorker,
+  }>()
+
+  const workerStatus = computed(() => props.worker.status)
+
+  const label = computed(() => {
+    return capitalize(props.worker.status)
+  })
```

Comment:
nit: I'm trying to reduce the number of computed properties we're using because they have some overhead associated with them. This is fine as is but you could also pass `worker.status` directly into `WorkerStatusIcon` and add `capitalize(worker.status)` directly to the template rather than using computed properties.